src/agent/src/image_rpc.rs

```diff
@@ -37,73 +38,58 @@ impl ImageService {
+        cid: &str,
+        source_creds: &Option<&str>,
+        policy_path: &Option<&String>,
+    ) -> Result<()> {
         let source_image = format!("{}{}", "docker://", image);
 
-        let mut manifest_path = PathBuf::from("/tmp");
-        manifest_path.push(cid);
-        manifest_path.push("image_manifest");
-        let target_path_manifest = format!("dir://{}", manifest_path.to_string_lossy());
-
-        // Define the target transport and path for the OCI image, without signature
-        let oci_path = Self::build_oci_path(cid);
+        let tmp_cid_path = Path::new("/tmp/").join(cid);
+        let oci_path = tmp_cid_path.join(IMAGE_OCI);
         let target_path_oci = format!("oci://{}", oci_path.to_string_lossy());
 
-        fs::create_dir_all(&manifest_path)?;
         fs::create_dir_all(&oci_path)?;
 
         info!(sl!(), "Attempting to pull image {}...", &source_image);
 
         let mut pull_command = Command::new(SKOPEO_PATH);
         pull_command
-            // TODO: need to create a proper policy
-            .arg("--insecure-policy")
             .arg("copy")
             .arg(source_image)
-            .arg(&target_path_manifest);
+            .arg(&target_path_oci)
+            .arg("--remove-signatures");
 
+        // If source credentials were passed (so not using an anonymous registry), pass them through
         if let Some(source_creds) = source_creds {
             pull_command.arg("--src-creds").arg(source_creds);
         }
 
-        let status: ExitStatus = pull_command.status()?;
-        ensure!(
-            status.success(),
-            "failed to copy image manifest: {:?}",
-            status,
-        );
-
-        // Copy image from one local file-system to another
-        // Resulting image is still stored in manifest format, but no longer includes the signature
-        // The image with a signature can then be unpacked into a bundle
-        let status: ExitStatus = Command::new(SKOPEO_PATH)
-            .arg("--insecure-policy")
-            .arg("copy")
-            .arg(&target_path_manifest)
-            .arg(&target_path_oci)
-            .arg("--remove-signatures")
-            .status()?;
+        // If a policy_path provided, use it, otherwise fall back to allow all image registries
+        if let Some(policy_path) = policy_path {
+            pull_command.arg("--policy").arg(policy_path);
+        } else {
+            pull_command.arg("--insecure-policy");
+        }
 
-        ensure!(status.success(), "failed to copy image oci: {:?}", status);
+        let status: ExitStatus = pull_command.status()?;
 
-        // To save space delete the manifest.
-        // TODO LATER - when verify image is added, this will need moving the end of that, if required
-        fs::remove_dir_all(&manifest_path)?;
+        if !status.success() {
+            fs::remove_dir_all(&tmp_cid_path)?;
+            return Err(anyhow!(format!("failed to pull image: {:?}", status)));
```

Comment:
This might be problematic: if we fail to remove the directory, the admin doesn't get a particularly meaningful message. Maybe you could save the error result of the `rmdir` and add it to the explicit return on the next line?

Reply:
I've attempted to do this, but it's pretty clumsy, so there is probably a nicer way